### PR TITLE
Spago docs --deps-only

### LINKED ISSUE #1020: Reinstate the `spago docs` command
We currently do not have a `spago docs` command in the PureScript codebase.

We would need to port it from the [old codebase](https://github.com/purescript/spago-legacy/blob/11e68ece50ede286660452ef370baa3ab5c33c02/src/Spago/Build.hs#L500), with the difference that since both projects are in PureScript we could just use it as a library instead of calling a separate node process.

So a tentative checklist for things to do could be:
- update [`docs-search`](https://github.com/purescript/purescript-docs-search) to use the new Spago
- add it as a dependency
- port [the implementation for the `docs` command from Haskell](https://github.com/purescript/spago-legacy/blob/11e68ece50ede286660452ef370baa3ab5c33c02/src/Spago/Build.hs#L500)
- since the `docs` command injects a Halogen app in the generated html, we'd need to bundle that app when bunling/publishing Spago and put it where Spago can find it at runtime, so it can copy it over

### PR DESCRIPTION
**Pull request by CharlesTaylor7**
### Description of the change

Resolves #1020 

With this change I believe we're basically at parity with the old spago docs command. 

(There was a --no-search flag in the old spago which skips indexing, but I don't see a strong reason to bring that back)

### Checklist:

- [ ] Added the change to the "Unreleased" section of the changelog
- [x] Added some example of the new feature to the `README`
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by CharlesTaylor7** [comment]
Technically the feature works, but the console outs give the impression it's building the project, even when isn't:

 ```console
 spago docs --deps-only
Reading Spago workspace configuration...
Read the package set from the registry

✅ Selecting 6 packages to build:
    docs-search-client-halogen
    docs-search-common
    docs-search-index
    spago
    spago-bin
    spago-core

Writing a new lockfile
Downloading dependencies...
Generating documentation for the project. This might take a while...
Documentation written to: generated-docs/html
Building the search index...
"bower.json" decoding failed for .spago/packages/avar-5.0.0/bower.json: An error occurred while decoding a JSON value:
  Expected value of type 'ROOT."main" - Expected a value of type Array but got String'.
Indexing 1004 modules from 127 packages...
Added 8064 definitions and 9636 type definitions from 127 packages to the search index.
Link: file:///Users/chuck/repos/spago/generated-docs/html/index.html
```

**Comment by CharlesTaylor7** [comment]
spago install also includes this extra terminal output:

```console
spago install
Reading Spago workspace configuration...
Read the package set from the registry

✅ Selecting 6 packages to build:
    docs-search-client-halogen
    docs-search-common
    docs-search-index
    spago
    spago-bin
    spago-core

Writing a new lockfile
Downloading dependencies...
Building...
           Src   Lib   All
Warnings     0     0     0
Errors       0     0     0

✅ Build succeeded.

```

File: README.md [hunk 1]
```diff
@@ -932,12 +932,19 @@ If you wish for the documentation to be opened in browser when generated, you ca
...
 
-To build the documentation as Markdown instead of HTML, or to generate tags for your project,
-you can pass a `format` flag:
+You can customize the output to other formats beyond html. Supported formats include ctags, etags, and markdown.
+For example to generate ctags for use in your editor:
 ```console
 $ spago docs --format ctags
 ```
 
+Sometimes you'd like to pull up docs for dependencies even when you have compilation errors in your project. This good use case for the --deps-only flag:
```
Comment (f-f) [inline README.md:941]:
```suggestion
Sometimes you'd like to pull up docs for dependencies even when you have compilation errors in your project - in such cases it's possible to use the --deps-only flag:
```

**Comment by f-f** [comment]
> There was a --no-search flag in the old spago which skips indexing, but I don't see a strong reason to bring that back

Agreed, don't think it's useful since indexing is pretty quick.

> the console output give the impression it's building the project, even when isn't

If it's not building the project then we should fix that, because we need to make sure the project is build before generating the index, otherwise the docs files will be out of date

> spago install also includes this extra terminal output

What's extra about it? Looks like regular business to me. If you're referring to the fact that it's building the project then sure, `spago install` does that (it's basically a `spago build --deps-only`)

**Comment by CharlesTaylor7** [comment]
> > the console output give the impression it's building the project, even when isn't
> 
> If it's not building the project then we should fix that, because we need to make sure the project is build before generating the index, otherwise the docs files will be out of date

No, I think its working as intended. 
 I just mean that it's building the dependencies of the projects, not the user's local code. To me, the message of "selecting 6 packages to build" sounds looks like its building the entire workspace. There's no indication it's going to stop short of building the application code until it stops.


> > spago install also includes this extra terminal output
> 
> What's extra about it? Looks like regular business to me. If you're referring to the fact that it's building the project then sure, `spago install` does that (it's basically a `spago build --deps-only`)


This is a good analogy. I'm thinking in both these cases it could alternately say something like "Selecting n packages to build (dependencies only)".

File: README.md (hunk 1 above)
Comment (CharlesTaylor7) [inline README.md:941]:
```suggestion
Sometimes you'd like to pull up docs for dependencies even when you have compilation errors in your project. This is a good use case for the --deps-only flag:
```

**Comment by f-f** [comment]
> I just mean that it's building the dependencies of the projects, not the user's local code. To me, the message of "selecting 6 packages to build" sounds looks like its building the entire workspace. There's no indication it's going to stop short of building the application code until it stops.

Ah, I see what you mean now. I think the point of saying that is to note to the user 'look, we found these packages in the workspace for our build', which makes sense in this case too, since we're kicking off a build - even if it's just to build the dependencies of the workspace, because we are going to figure them out from these packages.
So not sure how we could frase this any better

**Review by f-f — APPROVED** [review]

**Pull request merged**